meta/fraction-metadata/src/main/java/org/wildfly/swarm/fractions/scanner/FilePresenceScanner.java

```diff
@@ -22,12 +22,13 @@
 import java.util.function.Consumer;
 
 import org.wildfly.swarm.spi.meta.FileDetector;
+import org.wildfly.swarm.spi.meta.FileSource;
 import org.wildfly.swarm.spi.meta.FractionDetector;
 
 /**
  * @author Heiko Braun
  */
-public class FilePresenceScanner implements Scanner<String> {
+public class FilePresenceScanner implements Scanner<FileSource> {
     @Override
     public String extension() {
         return XML;
```

Comment:
I know you didn't touch this line and I don't want you to do anything about it, but I was wondering some time before if this is possibly wrong? I mean, we currently only detect XML files, so it doesn't restrict anything, but maybe it could in the future? WDYT? Just a comment, nothing more.